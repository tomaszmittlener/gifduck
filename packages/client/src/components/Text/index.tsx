import React, { FunctionComponent } from 'react'
import styled from 'styled-components'

interface TextProps {
  value: string
  color: string
}

const StyledText = styled.span`
  color: ${props => props.color};
  font-weight: ${props => props.theme.typography.weights.bold};
`

const Text: FunctionComponent<TextProps> = ({ value, color }) => {
  return <StyledText color={color}>{value}</StyledText>
}

export default Text
