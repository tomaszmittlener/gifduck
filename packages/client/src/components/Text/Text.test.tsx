import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import Theme from 'styles/theme'

import Text from './'

afterEach(cleanup)

describe('Text', () => {
  test('renders correctly with red color', async () => {
    const { container } = render(
      <ThemeProvider theme={Theme}>
        <Text value={'hello'} color={'red'} />
      </ThemeProvider>,
    )

    expect(container).toMatchSnapshot()
  })
  test('renders correctly with blue color', async () => {
    const { container } = render(
      <ThemeProvider theme={Theme}>
        <Text value={'hello'} color={'blue'} />
      </ThemeProvider>,
    )

    expect(container).toMatchSnapshot()
  })
})
