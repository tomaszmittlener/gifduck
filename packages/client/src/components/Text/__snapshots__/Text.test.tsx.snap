// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly with blue color 1`] = `
.c0 {
  color: blue;
  font-weight: 700;
}

<div>
  <span
    class="c0"
    color="blue"
  >
    hello
  </span>
</div>
`;

exports[`Text renders correctly with red color 1`] = `
.c0 {
  color: red;
  font-weight: 700;
}

<div>
  <span
    class="c0"
    color="red"
  >
    hello
  </span>
</div>
`;
